tools/nodejs_api/src_nodejs/queryResult.js

```diff
@@ -0,0 +1,47 @@
+const callbackWrapper = require("./common.js");
+
+class QueryResult {
+    #queryResult
+    #isClosed
+    constructor(queryResult) {
+        this.#queryResult = queryResult;
+        this.#isClosed = false;
+    }
+
+    checkForQueryResultClose(){
+        if (this.#isClosed){
+            throw new Error("Query Result is Closed");
+        }
+    }
+
+    close(){
+        if (this.#isClosed){
+            return;
+        }
+        this.#queryResult.close();
+        this.#isClosed = true;
+    }
+
+    each(eachCallback, doneCallback) {
+        this.checkForQueryResultClose();
+        this.#queryResult.each((err, result) => {
```

Comment:
The semantic is not correct. You should pass both the `eachCallbak` and `doneCallback` to C++ side.